HousingSearchApi/V2/Controllers/SearchController.cs

```diff
@@ -0,0 +1,43 @@
+using Amazon.Lambda.Core;
+using Hackney.Core.Logging;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using HousingSearchApi.V2.Domain.DTOs;
+using HousingSearchApi.V2.UseCase;
+
+namespace HousingSearchApi.V2.Controllers;
+
+[ApiVersion("1")]
+[Produces("application/json")]
+[Route("api/v2/search/")]
+[ApiController]
+public class SearchController : Controller
+{
+    private readonly SearchUseCase _searchUseCase;
+
+    public SearchController(SearchUseCase searchUseCase)
+    {
+        _searchUseCase = searchUseCase;
+    }
+
+    [LogCall(Microsoft.Extensions.Logging.LogLevel.Information)]
+    [HttpGet("{indexName}")]
+    public async Task<IActionResult> Search(string indexName, [FromQuery] SearchParametersDto searchParametersDto)
+    {
+        var searchResults = await _searchUseCase.ExecuteAsync(indexName, searchParametersDto).ConfigureAwait(false);
+
+        var response = new
+        {
+            Results = new Dictionary<string, IReadOnlyCollection<object>>
+            {
+                [indexName] = searchResults.Documents
+            },
```

Comment:
I guess this wrapper within the wrapper is a compatibility thing?

Reply:
Yep - all the other endpoints return in this kind of structure